### PR TITLE
fix: gitignoring glob behavior introduced in #1234

### PR DESCRIPTION
**Pull request by cakekindel**
see #1235

### CODE REVIEW

**Comment by cakekindel** [comment]
opening this PR now so I can run against the windows CI container

**Comment by cakekindel** [comment]
I was able to reproduce with wine actually, should have a fix soon

File: src/Spago/Glob.purs [hunk 1]
```diff
@@ -49,7 +49,7 @@ gitignoreGlob base =
       ( \line -> do
           let
             resolve a = Path.concat [ base, a ]
-            pat a = resolve $ unpackPattern a
+            pat a = withForwardSlashes $ resolve $ unpackPattern a
```
Comment (cakekindel) [inline src/Spago/Glob.purs:52]: :melting_face:
`Path.concat ["foo\\bar", "**"] == "foo\\bar\\**"`
`\**` and `/**` are ***very*** different.

File: test/Spago/Glob.purs [hunk 2]
```diff
@@ -48,36 +50,55 @@ globTmpDir m = Aff.bracket make cleanup m
...
-
-      Spec.it "respects some .gitignore patterns" \p -> do
-        FS.writeTextFile (Path.concat [p, ".gitignore"]) """/fruits\n/src"""
-        a <- Glob.gitignoringGlob p ["fruits/apple/**"]
-        a `Assert.shouldEqual` ["fruits/apple"]
+            a <- Glob.gitignoringGlob p ["fruits/**/apple"]
+            Array.sort a `Assert.shouldEqual` ["fruits/left/apple", "fruits/right/apple"]
 
       Spec.it "is stacksafe" \p -> do
-        hugeGitignore <- liftEffect $ intercalate "\n" <$> randomSample' 10000 (resize 4 $ genAlphaLowercaseString)
```
Comment (cakekindel) [inline test/Spago/Glob.purs:80]: this would sometimes randomly generate `left` which would fail the test :grin:

**Comment by cakekindel** [comment]
@f-f this is ready!

**Review by f-f — APPROVED** [review]
Ah, amazing. Thank you ❤️

**Pull request merged**